midp/gfx.js

```diff
@@ -112,8 +112,20 @@ var currentlyFocusedTextEditor;
         return 1;
     };
 
-
-    addUnimplementedNative("com/sun/midp/lcdui/DisplayDevice.refresh0.(IIIIII)V");
+    Native["com/sun/midp/lcdui/DisplayDevice.refresh0.(IIIIII)V"] = function() {
+        // This is a no-op: The foreground display gets drawn directly
+        // to the screen.
+        //
+        // However, we sometimes "draw" more frequently than the
+        // screen can update. By calling `requestAnimationFrame` here,
+        // we ensure that whatever we just drew will stick around for
+        // one frame
+        asyncImpl("V", new Promise(function(resolve, reject) {
```

Comment:
Maybe in this case we shouldn't use asyncImpl (otherwise the thread would be resumed after a setZeroTimeout and after a requestAnimationFrame, but we want it to be resumed only after requestAnimationFrame).